### ISSUE TITLE
CompilationUnit#getFilePath fails in multimodule project build with XtextBuild in Eclipse

### ISSUE THREAD

**Issue by olga-kr** [eclipse/xtext-xtend#383]
There's custom active annotation processor that adds nested class to the processed class. If annotatad class B extends annotated class A, then B.Inner must extend A.Inner. Classes A and B reside in different modules.
Everything works as expected. 
Then I add CompilationUnit#getFilePath call to my processor and build class B with eclipse tools. Compilation ends successfully but  B.Inner do not extend A.Inner anymore. If the same is done via maven everythings is still ok. So there are several points:
1. If CompilationUnit#getFilePath is not supposed to work in this case it has to be documented.
2. It seems strange that compilation proceeds after fail on CompilationUnit#getFilePath and produces java file which is in unexpected state
3. I hardly managed to get stacktrace (print it to file eventually). Where should i look in eclipse to find this kind of errors?

Stacktrace and test project in attachment.

**Comment by olga-kr** [comment]
[context_test.zip](https://github.com/eclipse/xtext-xtend/files/1815797/context_test.zip)

**Comment by cdietrich** [comment]
```
java.lang.IllegalArgumentException: The given path '/resource/context_module_a/src/main/java/org/test/A.xtend' is not relative.
	at org.eclipse.xtend.lib.macro.file.Path.getAbsolutePath(Path.java:169)
	at org.eclipse.xtend.lib.macro.file.Path.getAbsolutePath(Path.java:160)
	at org.eclipse.xtend.core.macro.AbstractFileSystemSupport.getPath(AbstractFileSystemSupport.java:468)
	at org.eclipse.xtend.core.macro.AbstractFileSystemSupport.getPath(AbstractFileSystemSupport.java:458)
	at org.eclipse.xtend.core.macro.AbstractFileSystemSupport.getPath(AbstractFileSystemSupport.java:450)
	at org.eclipse.xtend.core.macro.declaration.CompilationUnitImpl.getFilePath(CompilationUnitImpl.java:391)
	at org.test.AnnProcessor.doRegisterGlobals(AnnProcessor.java:23)
	at org.eclipse.xtend.lib.macro.AbstractClassProcessor.doRegisterG
```

**Comment by cdietrich** [comment]
```
JdtProjectConfig [
  project = P/context_module_b
  projectConfigProvider = org.eclipse.xtext.ui.workspace.EclipseProjectConfigProvider@7339f0a
]
```

protected def getPath(URI absoluteURI, URI baseURI, Path basePath) {
- platform:/resource/context_module_a/src/main/java/org/test/A.xtend
- platform:/resource/context_module_b/
- /context_module_b

**Comment by cdietrich** [comment]
=> its intended to work but does not work with multi project cause path calculation explodes

**Comment by cdietrich** [comment]
i wonder if something like this would do the trick

```
	protected def getPath(URI absoluteURI, URI baseURI, Path basePath) {
		val URI relativeURI = 
		if (baseURI.isPlatformResource && absoluteURI.isPlatformResource) {
			if (baseURI.segment(1) != absoluteURI.segment(1)) {
				val p = new org.eclipse.core.runtime.Path(absoluteURI.toPlatformString(true));
				URI.createURI(".."+p.toString)
			} else {	
				absoluteURI.deresolve(baseURI)
			}
		} else {
			absoluteURI.deresolve(baseURI)
		}
		 
		if (relativeURI.empty || relativeURI == absoluteURI)
			return null
		
		basePath.getAbsolutePath(relativeURI.toString)
	}
```

**Comment by olga-kr** [comment]
Sorry, I don't get how to apply this method to CompilationUnit object

**Comment by cdietrich** [comment]
This is the patch for Xtext. You cannot do anything

**Comment by cdietrich** [comment]
@szarnekow what do you think? do you know any better api?

**Comment by szarnekow** [comment]
I'm not sure yet. I have to better understand the underlying problem. What is the difference between 
```
val p = new org.eclipse.core.runtime.Path(absoluteURI.toPlatformString(true));
URI.createURI(".."+p.toString)
```
and
```
val p = absoluteURI.toPlatformString(true);
URI.createURI("../"+p)
```?

**Comment by cdietrich** [comment]
fixed in 2.14.0(SNAPSHOT/Nighlty) @olga-kr could you give it a try (http://download.eclipse.org/modeling/tmf/xtext/updates/composite/latest/)

**Comment by olga-kr** [comment]
Yes, it helped. Thanks a lot.
